src/igvfd/types/file.py

```diff
@@ -363,32 +367,35 @@ def upload_credentials(self):
     def create(cls, registry, uuid, properties, sheets=None):
         if properties.get('upload_status') == 'pending':
             sheets = {} if sheets is None else sheets.copy()
-            if properties.get('controlled_access') is True:
-                bucket = registry.settings['restricted_file_upload_bucket']
-                sts_client = get_restricted_sts_client(
-                    localstack_endpoint_url=os.environ.get(
-                        'LOCALSTACK_ENDPOINT_URL'
-                    )
-                )
+            if properties.get('externally_hosted') is True:
```

Comment:
I think we want this check at the top, before `if properties.get('upload_status') == 'pending':`